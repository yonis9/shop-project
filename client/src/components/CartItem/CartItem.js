import React from 'react';

import {
    CartItemContainer,
    ImageContainer,
    ItemDetailsContainer
 } from './CarItem.styles';

const CartItem = ({ item: { name, imageUrl, price, quantity } }) => (
    <CartItemContainer>
        <ImageContainer src={imageUrl} alt='item' />
        <ItemDetailsContainer>
            <span>{name}</span>
            <span >{quantity} x {price}</span>
        </ItemDetailsContainer>
    </CartItemContainer>
)

export default React.memo(CartItem);