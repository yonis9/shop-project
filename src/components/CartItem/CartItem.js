import React from 'react';

import {
    CartItemContainer,
    ImageContainer,
    ItemDetailsContainer,
    NameContainer
 } from './CarItem.styles';

const CartItem = ({ item: { name, imageUrl, price, quantity}}) => (
    <CartItemContainer>
        <ImageContainer src={imageUrl} alt='item' />
        <ItemDetailsContainer>
            <span>{name}</span>
            <span >{quantity} x {price}</span>
        </ItemDetailsContainer>
    </CartItemContainer>
)

export default CartItem;