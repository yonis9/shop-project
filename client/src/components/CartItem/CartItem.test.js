import React from 'react';
import { shallow } from 'enzyme';

import CartItem from './CartItem';

it('CartItem component', () => {
    const mockItem = {
        name: 'shirt',
        imageUrl: 'shirts.com',
        price: 15,
        quantity: 2
    }

    expect(shallow(<CartItem item={mockItem} />)).toMatchSnapshot();
})