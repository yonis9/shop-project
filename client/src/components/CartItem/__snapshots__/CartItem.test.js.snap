// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component 1`] = `ShallowWrapper {}`;
